main.go

```diff
@@ -99,10 +100,14 @@ func entrypoint(shutdown chan os.Signal) {
 	)
 
 	registry := prometheus.NewRegistry()
-	sdk, err := core.NewSDK(ctx, logrus.NewEntry(log), oas, opaModuleConfig, &core.EvaluatorOptions{
-		EnablePrintStatements: env.IsTraceLogLevel(),
-		MongoClient:           mongoClient,
-	}, registry, env.ClientTypeHeader)
+	sdk, err := sdk.NewFromOAS(ctx, opaModuleConfig, oas, &sdk.FromOASOptions{
```

Comment:
I think we definitely reached a nice API 👍 

(maybe options could be collapsed in a single struct with everything in it? could be messy to document?)